frontend/javascripts/oxalis/view/action-bar/download_modal_view.tsx

```diff
@@ -0,0 +1,332 @@
+    modalType = tab;
+    console.log(modalType);
+  };
+
+  const handleLayerSelection = (selection: string) => {
+    console.log(selection);
+  };
+
+  const handleBoundingBoxSelection = (bbox: string) => {
+    console.log(bbox);
+  };
+
+  const checkboxStyle = {
+    height: "30px",
+    lineHeight: "30px",
+  };
+  const selection = ["Volume", "Skeleton", "Fallback"];
+
+  const moreInfoHint = (
+    <Hint
+      style={{
+        margin: "0px 12px 0px 12px",
+      }}
+    >
+      For more information on how to process downloaded layers visit the{" "}
+      <a href="https://docs.webknossos.org" target="_blank" rel="noreferrer">
+        user documentation...
+      </a>
+    </Hint>
+  );
+
+  const wklibsInitCode = `import webknossos as wk
+
+with wk.webknossos_context(token="MY_TOKEN"):
+    dataset = wk.Dataset.download(
+        dataset_name_or_url="dataset_name",
+        organization_id="my_organization",
+    )
+`;
+
+  return (
+    <Modal
+      title="Download this Annotation"
+      visible={isVisible}
+      width={600}
+      okText="Ok"
+      onOk={handleOk}
+      onCancel={onOk}
+      style={{ overflow: "visible" }}
+    >
+      <Tabs onChange={handleTabChange} type="card">
+        <TabPane tab="Download" key="1">
+          <Row>
+            <Col span={24}>
+              <Hint
+                style={{
+                  margin: "6px 12px",
+                }}
+              >
+                {messages["annotation.download"]}
+              </Hint>
+            </Col>
+          </Row>
+          <Divider
+            style={{
+              margin: "18px 0",
+            }}
+          >
+            Options
+          </Divider>
+          {maybeShowWarning()}
+          <Row>
+            <Col
+              span={9}
+              style={{
+                lineHeight: "20px",
+                padding: "5px 12px",
+              }}
+            >
+              Select the data you would like to download.
+            </Col>
+            <Col span={15}>
+              <CheckboxGroup onChange={handleCheckboxChange} defaultValue={selection}>
+                <Checkbox style={checkboxStyle} value="Volume">
+                  Volume as WKW
+                </Checkbox>
+                <Hint
+                  style={{
+                    marginLeft: 24,
+                    marginBottom: 12,
+                  }}
+                >
+                  Download a zip folder containing WKW files.
+                </Hint>
+
+                <Checkbox style={checkboxStyle} value="Skeleton">
+                  Skeleton as NML
+                </Checkbox>
+                <Hint
+                  style={{
+                    marginLeft: 24,
+                    marginBottom: 12,
+                  }}
+                >
+                  Download a zip folder containing NML files.
+                </Hint>
+
+                <Checkbox style={checkboxStyle} value="Fallback">
+                  Fallback Layers
```

Comment:
This should only be selectable / enabled if a fallback layer is part of the DS.

Reply:
Also I am not quite sure if I understand this feature. This would likely download a whole segmentation layer, right?

Reply:
This was a suggestion from me (not fully understanding the extent that fallback layers could have) and will be removed.

Reply:
The exception being the tiff export job (webknossos-worker feature). That one does actually merge the volume annotation with the fallback layer. The trick here is that the user has to select a small-ish bounding box for the tiff export so that they do not create a huge load for job and download. This limit is currently set to 1024 Mvx. (can be configured in application.conf and queried in api/features)